pkg/registry/scripts/projects.js

```diff
@@ -128,7 +167,13 @@
                 var display = fields.display.trim();
                 var description = fields.description.trim();
 
-                if (!NAME_RE.test(name)) {
+                if (!name) {
```

Comment:
Alright. Makes sense. But in that case please refactor this code so that it's not a big set of copy paste. We should be creating Errors based on conditions, and if an error is set, have one block of code to create the deferred.

In addition this, most of code should be factored out into a function, called validate-name that returns an exception ... so that it's not duplicated between all these dialogs. Because you're now combining all of these into one projectActions block, that should be easy.